lib/std/enum_set.zig

```diff
@@ -0,0 +1,144 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+const std = @import("std.zig");
+const assert = std.debug.assert;
+const testing = std.testing;
+
+pub fn EnumSet(comptime Enum: type) type {
```

Comment:
One thought I had was: should I add a `options` argument for future extensions?

e.g. I can imagine in future wanting to specify endieness for the `PackedIntArray`